src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,494 @@
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(closestValueIndex);
+          pointerValue = getBoundedValue(pointerValue, minVal, maxVal);
+          if (pointerValue === currentValues[closestValueIndex]) {
+            return;
+          }
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange?.(newValues);
+          focusThumb(containerRef.current, closestValueIndex);
+          event.preventDefault();
+          event.stopPropagation();
+        }
+      },
+      [min, max, step, currentValues, getAllowableThumbRange, onChange],
+    );
+
+    const pointerMoveFunctionRef = React.useRef<
+      (event: PointerEvent) => void
+    >();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerMoveFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener(
+          'pointermove',
+          pointerMoveFunctionRef.current,
+        );
+      }
+      pointerMoveFunctionRef.current = handlePointerMove;
+    }, [handlePointerMove]);
+
+    const pointerUpFunctionRef = React.useRef<(event: PointerEvent) => void>();
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      if (pointerUpFunctionRef.current && ownerDoc) {
+        ownerDoc.removeEventListener('pointerup', pointerUpFunctionRef.current);
+      }
+      pointerUpFunctionRef.current = handlePointerUp;
+    }, [handlePointerUp]);
+
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      ownerDoc?.addEventListener('pointermove', handlePointerMove);
+      ownerDoc?.addEventListener('pointerup', handlePointerUp);
+      return () => {
+        ownerDoc?.removeEventListener('pointermove', handlePointerMove);
+        ownerDoc?.removeEventListener('pointerup', handlePointerUp);
+      };
+    }, [handlePointerMove, handlePointerUp]);
+
+    const tickMarkArea = React.useMemo(() => {
+      if (!tickLabels) {
+        return null;
+      }
+
+      if (Array.isArray(tickLabels)) {
+        return (
+          <div className='iui-slider-ticks'>
+            {tickLabels.map((label, index) => (
+              <span key={index} className='iui-slider-tick'>
+                {label}
+              </span>
+            ))}
+          </div>
+        );
+      }
+
+      return tickLabels;
+    }, [tickLabels]);
+
+    return (
+      <div
+        ref={ref}
+        className={cx(
+          'iui-slider-component-container',
+          { 'iui-disabled': disabled },
+          className,
+        )}
+        {...rest}
+      >
+        {minValueLabel && (
+          <span className='iui-slider-min'>{minValueLabel}</span>
+        )}
+        <div
+          ref={containerRef}
+          className={cx('iui-slider-container', {
+            'iui-grabbing': undefined !== activeThumbIndex,
```

Comment:
test to check if iui-grabbing is applied when needed

Reply:
done